src/System.Text.Primitives/System/Text/Parsing/PrimitiveParserHelpers.t4

```diff
@@ -0,0 +1,197 @@
+﻿<#@ import namespace="System.Collections.Generic" #>
+<#@ import namespace="System.Text" #>
+<#+
+enum BufferType
+{
+	ArrayPointer,
+	ReadOnlySpan
+}
+
+enum Encoding
```

Comment:
Ah, I see. In the previous comment I assumed the types were our existing encoding types, but you have defined new encoding enum. I think it might be better to change this name to avoid confusion and clashes with System.Text.Encoding.